sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/objects/objects.scala

```diff
@@ -1787,3 +1787,75 @@ case class ValidateExternalType(child: Expression, expected: DataType)
     ev.copy(code = code, isNull = input.isNull)
   }
 }
+
+object GetArrayFromMap {
+  abstract class Source
+  case class Key() extends Source
+  case class Value() extends Source
```

Comment:
Thought, this adds readability. Is a simple boolean param better?

Reply:
I don't have a strong opinion, but shall we use enum?

Reply:
Yeah, probably.